src/internal/connector/exchange/api/shared.go

```diff
@@ -104,19 +159,29 @@ func getItemsAddedAndRemovedFromContainer(
 			}
 		}
 
-		nextLink, delta := api.NextAndDeltaLink(resp)
+		dresp, ok := resp.(api.DeltaPageLinker)
+		if ok {
+			nextLink, deltaLink = api.NextAndDeltaLink(dresp)
+		} else {
+			nextLink = api.NextLink(resp)
+			deltaLink = "" // to make sure we don't use an old value
+		}
```

Comment:
Still concerned about side-loading interfaces.  Seems like this could be changed so that all responses are DeltaPageLinkers, and in the case of non-delta queries the deltaLink is always an empty string.

Reply:
Continuation of https://github.com/alcionai/corso/pull/3212#discussion_r1176697616

> Seems like this could be changed so that all responses are DeltaPageLinkers, and in the case of non-delta queries the deltaLink is always an empty string.

Unfortunately that is not possible as the object that is returned from non-delta calls does not have `GetOdataDeltaLink` and so cannot satisfy [`DeltaPageLinker`](https://github.com/alcionai/corso/blob/1b417af5bb1e6f250adb7e21216a731e9decb909/src/internal/connector/graph/api/api.go#L13).

> ... we should probably move NextAndDeltaLink to be a function on the pageLinker instead of a function that takes pageLinker

That would be better, but as we can't satisfy the interface, it will still be moving the check to within the function as done [here](https://github.com/alcionai/corso/pull/3212#discussion_r1176697616) unless I am mistaken.

Reply:
> Unfortunately that is not possible as the object that is returned from non-delta calls does not have GetOdataDeltaLink and so cannot satisfy [DeltaPageLinker](https://github.com/alcionai/corso/blob/1b417af5bb1e6f250adb7e21216a731e9decb909/src/internal/connector/graph/api/api.go#L13).

You can wrap the object to normalize the interface.  We do [similar things with calendar containers in graph](https://github.com/alcionai/corso/blob/main/src/internal/connector/graph/cache_container.go#L156).